server/app/api.py

```diff
@@ -1987,6 +2010,72 @@ def new_entity(self, attributes):
         ent.assigned_staff = [models.User.get_or_insert(
             user.id()).key for user in ent.assigned_staff]
         return ent
+    
+    
+class QueuesAPI(APIResource):
+    """ API resource for sets of queues """
+    
+    contains_entities = False
+
+    methods = {
+        'generate': {
+            'methods': set(['POST']),
+            'web_args': {
+                'course': KeyArg('Course', required=True),
+                'assignment': KeyArg('Assignment', required=True)
+            }
+        }
+    }
+    
+    def generate(self, user, data):
+        """ Splits up submissions among staff members """
+        
+        self.check_permissions(user, data)
+        course, assignment = data['course'].get(), data['assignment'].get()
+        userify = lambda parts: [part.user.get() for part in parts]
+        
+        staff = userify(models.Participant.query(
+            models.Participant.role == STAFF_ROLE,
+            models.Participant.course == course.key).fetch())
+
+        students = userify(models.Participant.query(
+            models.Participant.role == STUDENT_ROLE,
+            models.Participant.course == course.key).fetch())
+
+        subms, queues = [], []
+
+        for student in students:
```

Comment:
Wouldn't this double submit into a queue if a student is part of a group? I think it'd be better to pull from FinalSubmissions.